diff --git a/configure b/configure
index ebe3d8c..a336b73 100755
--- a/configure
+++ b/configure
@@ -18943,7 +18943,6 @@ do :
   ac_fn_c_check_func "$LINENO" "sendfile" "ac_cv_func_sendfile"
 if test "x$ac_cv_func_sendfile" = xyes; then :
   cat >>confdefs.h <<_ACEOF
-#define HAVE_SENDFILE 1
 _ACEOF

 fi
diff --git a/io.c b/io.c
index 48592ac51a..2635ac8ec6 100644
--- a/io.c
+++ b/io.c
@@ -11111,6 +11111,7 @@ nogvl_copy_stream_wait_write(struct copy_stream_struct *stp)
     return 0;
 }

+#undef USE_COPY_FILE_RANGE
 #ifdef USE_COPY_FILE_RANGE

 static ssize_t
